src/Engine.sol

```diff
@@ -401,20 +494,28 @@ contract Engine is
         external
         override
     {
-        // ensure account exists (i.e. owner is not the zero address)
-        /// @notice this does not check if the caller is the account owner
-        if (PERPS_MARKET_PROXY.getAccountOwner(_accountId) == address(0)) {
-            revert AccountDoesNotExist();
-        }
-
         credit[_accountId] += _amount;
 
-        /// @dev sUSD transfers that fail will revert
+        /// @dev $sUSD transfers that fail will revert
         SUSD.transferFrom(msg.sender, address(this), _amount);
 
         emit Credited(_accountId, _amount);
     }
 
+    /// @inheritdoc IEngine
+    function creditAccountZap(
+        uint128 _accountId,
+        uint256 _amount,
+        address _referrer
+    ) external override {
+        // zap $USDC -> $sUSD
+        uint256 usdcAmount = _zapIn(_amount, _referrer);
+
+        credit[_accountId] += usdcAmount;
```

Comment:
**Nitpick:** This is really a non-issue, but in `creditAccount`, you increment storage before executing `transferFrom`, following an effects-interactions pattern. Here you `transferFrom`/`_zapIn` before incrementing storage, so it goes interactions-effects. In this case there are no checks anyway and it doesn't seem to cause any security issue, but it bugged me (even just from a consistency/following the same coding pattern perspective), so I thought I would mention it. But can't see any security issue.

Reply:
changed

Reply:
**Question:** Did you want to go ahead with this change. Looking at the code now and I don't think it has changed?